src/idl_gen_swift.cpp

```diff
@@ -9,30 +9,34 @@ namespace flatbuffers {
 
 namespace swift {
 
-inline std::string GenIndirect(std::string reading) {
+inline std::string GenIndirect(const std::string reading) {
```

Comment:
It should be const reference: `const std::string &reading`.
The same for others `const std::string`, they should be declared as reference `const std::string&`.

`const std::string` is a pass-by-value semantic, if inlining failed then a copy of string will be created.
With reference, a compiler will pass the address of a string.

Reply:
@vglavnyy so if we inline a function in cpp, we should use references instead of just passing a std::string.
I’m still kinda learning cpp so forgive me for that

Reply:
It doesn't depend on inlining. Inline word is a recommendation only which can be ignored by a compiler.
```cpp
void foo(const std::string s) {
 // s is immutable string
}

void bar(const std::string s) {
 // s is reference to immutable string
}

void test() {
 std::string text = "bar and foo";
 foo(text); // foo(std::string(text)) - text will be cloned and passed into foo
 bar(text); // bar(&text) - pass address of text without clone
}
```

Reply:
Thanks for the explanation!